src/instance.rs

```diff
@@ -172,18 +182,6 @@ impl<'py, T> Bound<'py, T> {
         self.into_non_null().as_ptr()
     }
 
-    /// Internal helper to convert e.g. &'a &'py PyDict to &'a Bound<'py, PyDict> for
-    /// backwards-compatibility during migration to removal of pool.
-    #[doc(hidden)] // public and doc(hidden) to use in examples and tests for now
-    pub fn borrowed_from_gil_ref<'a, U>(gil_ref: &'a &'py U) -> &'a Self
-    where
-        U: PyNativeType<AsRefSource = T>,
-    {
-        // Safety: &'py T::AsRefTarget is expected to be a Python pointer,
-        // so &'a &'py T::AsRefTarget has the same layout as &'a Bound<'py, T>
-        unsafe { std::mem::transmute(gil_ref) }
-    }
-
     /// Internal helper to get to pool references for backwards compatibility
     #[doc(hidden)] // public and doc(hidden) to use in examples and tests for now
```

Comment:
I think we also need to make these into proper public API for the reverse direction?

Reply:
Sure thing, it is definitely reasonable to have them even even if they have clunky names and we encourage users to use `.as_bound()` / `.as_borrowed()`.

Reply:
> we encourage users to use .as_bound() / .as_borrowed().

I am possibly missing something. I though these are not redundant to `as_bound` (GIL -> bound) as they provide the reverse direction (bound -> GIL)?

Reply:
Sorry my mistake, from the location of the comment I thought earlier this was suggesting to make `Bound::borrowed_from_gil_ref` public API instead of deleting it. I see now that this is suggesting to make `.as_gil_ref()` public; we _definitely_ want to do that!